official/transformer/v2/metrics.py

```diff
@@ -160,9 +160,10 @@ def get_config(self):
 
   def call(self, inputs):
     logits, targets = inputs[0], inputs[1]
-    for mean, fn in self.metric_mean_fns:
-      m = mean(*fn(logits, targets))
-      self.add_metric(m)
+    if not tf.distribute.in_cross_replica_context():
```

Comment:
Leave a comment and bug link if relevant as to why this is needed, it seems odd that we have to do something special.

Reply:
Found another way to write metrics that doesn't need this hack. Bug filed to track the issue with the old approach.